code/controllers/subsystem/weeds.dm

```diff
@@ -17,89 +19,90 @@ SUBSYSTEM_DEF(weeds)
 		currentrun = pending.Copy()
 		creating = list()
 
-	for(var/A in currentrun)
+	for(var/turf/T AS in currentrun)
 		if(MC_TICK_CHECK)
 			return
 
-		var/obj/effect/alien/weeds/node/N = currentrun[A]
-		currentrun -= A
-		var/turf/T = A
+		var/obj/effect/alien/weeds/node/node = currentrun[T]
+		currentrun -= T
 
-		if(QDELETED(N) || QDELETED(T))
+		if(QDELETED(node) || QDELETED(T))
 			pending -= T
+			attempt -= T
 			continue
 
-		if ((locate(/obj/effect/alien/weeds) in T) || (locate(/obj/effect/alien/weeds/node) in T))
+		if (locate(/obj/effect/alien/weeds/node) in T)
 			pending -= T
+			attempt -= T
 			continue
 
 		if (!T.is_weedable())
 			pending -= T
+			attempt -= T
 			continue
 
 		for(var/direction in GLOB.cardinals)
 			var/turf/AdjT = get_step(T, direction)
-			if (!(AdjT in N.node_turfs)) // only count our weed graph as eligble
-				continue
```

Comment:
this created jittery in the creation of the weeds. It's not really needed with the new get_dist system